src/puppetlabs/puppetdb/http/fact_names.clj

```diff
@@ -14,27 +14,26 @@
    paging parameters. Also accepts GETs and POSTs. Composes
    `optional-handlers` with the middleware function that executes the
    query."
-  [entity version param-spec & optional-handlers]
-  (app
-    (http-q/extract-query param-spec)
-    (apply comp
-           (fn [{:keys [params globals puppetdb-query]}]
-             (let [puppetdb-query (assoc-when puppetdb-query :order_by [[:name :ascending]])]
-               (produce-streaming-body
-                 entity
+  [version param-spec & optional-handlers]
+  (let [handlers (cons (partial http-q/restrict-query-to-entity "fact_names")
```

Comment:
Why not omit the let (and the cons) and just add the partial and optional handlers in to the comp below?